### PR TITLE
[feature]: css and @testing-library/react integration to the editor.

### PR DESCRIPTION
**Pull request by fayzzzm**

### CODE REVIEW

**Comment by fayzzzm** [comment]
Some of the parts need to be refactored and written tests for each case. As soon as possible, I cover all of the needed parts with tests and also need to config the webpack and other helpers to the advanced level. Need further instructions from you @VoIdJMp.

**Comment by fayzzzm** [comment]
Don't worry about CI failing. I will fix it.

File: src/server/src/routers/file-io/providers/read-file/index.ts [hunk 1]
```diff
@@ -2,19 +2,38 @@ import { Request, Response } from 'express';
...
+            return 'index.html';
+        case 'css':
+            return 'style.css';
+        default:
+            throw new Error(
+                `This server doesn't support this extension: "${extension}". Please be sure that you are providing extensions such as: "html", "css" or "javascript"!`,
+            );
+    }
+};
+
```
Comment (VoIdJMp) [inline src/server/src/routers/file-io/providers/read-file/index.ts:19]: This function is used in several places. Can we add it to the global function?

File: src/client/src/scenes/Editor/index.tsx [hunk 2]
```diff
@@ -59,6 +59,8 @@ export const Editor: React.FC<IEditor> = inject('editorDataModel')(
             }
         };
 
+        console.log(extension);
+
```
Comment (VoIdJMp) [inline src/client/src/scenes/Editor/index.tsx:63]: Will this console.log go to prod?

**Comment by VoIdJMp** [comment]
```
jobs:
  build:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [12.x, 14.x, 16.x]
        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/

    steps:
    - uses: actions/checkout@v2
    - name: Use Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v2
      with:
        node-version: ${{ matrix.node-version }}
        cache: 'npm'
    - run: npm ci
    - name: Deploy
      run: |
        git config --global user.name $user_name
        git config --global user.email $user_email
        git remote set-url origin https://${github_token}@github.com/${repository}
        npm run deploy
      env:
        user_name: 'github-actions[bot]'
        user_email: 'github-actions[bot]@users.noreply.github.com'
        github_token: ${{ secrets.ACTIONS_DEPLOY_ACCESS_TOKEN }}
        repository: ${{ github.repository }}
```



You also need to add an auto-deploy to CI

File: src/client/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+module.exports = {
+    roots: ['<rootDir>/src'],
+    transform: {
+        '^.+\\.tsx?$': 'ts-jest',
+    },
+    testRegex: '(/__tests__/.*|(\\.|/)(test|spec))\\.tsx?$',
+    moduleNameMapper: {
+        '^@app/(.*)$': '<rootDir>/src/$1',
+        '\\.(css|scss)$': '<rootDir>/styleMock.js',
+    },
```
Comment (fayzzzm) [inline src/client/jest.config.js:10]: to resolve the aliases that we have. And also, in the second part, I've written an alias for matching `css | scss` files in order to mock them. Jest can't resolve the other files except for javascript files.

File: src/client/src/components/FileButton/__test__/index.test.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    );
+
+    expect(container.firstChild).toHaveClass('file-button');
+
+    // fireEvent.click(container.firstChild);
+    // expect(setActiveIndex).toBeCalledTimes(1);
+
+    // fireEvent.click(container.lastChild);
+    // expect(setActiveIndex).toBeCalledTimes(2);
+});
```
Comment (fayzzzm) [inline src/client/src/components/FileButton/__test__/index.test.tsx:56]: It is the base test that I've written for testing how does it work in a custom webpack config. Seems to work well. Moving forward to do some fancy stuff :)

File: src/client/src/scenes/Editor/index.tsx (hunk 2 above)
Comment (fayzzzm) [reply src/client/src/scenes/Editor/index.tsx:63]: No. For now, we have a problem with our linter, but as soon as possible, I will fix for pre-commit script it should check whether I don't have any `console.*`.

File: src/server/src/routers/file-io/providers/write-file.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { Request, Response } from 'express';
+import * as fs from 'fs';
+import * as path from 'path';
+
+import { getFileNameByExtension } from '../../../utils/getFileName';
```
Comment (fayzzzm) [inline src/server/src/routers/file-io/providers/write-file.ts:5]: Need to use webpack for bundling, or we can create a node-modules folder in order to trick node to find our aliased folder.

**Review by VoIdJMp — APPROVED** [review]

File: src/client/src/components/FileButton/__test__/index.test.tsx (hunk 4 above)
Comment (fayzzzm) [reply src/client/src/components/FileButton/__test__/index.test.tsx:56]: Done. The editor, for now, has a test case where the component is fully covered with the test. Moving forward to cover all components with tests and coverage with our rules.

File: src/client/src/scenes/Editor/index.tsx (hunk 2 above)
Comment (fayzzzm) [reply src/client/src/scenes/Editor/index.tsx:63]: @VoIdJMp Moving forward to advance other configs for pre-push trash code prevent.

**Pull request merged**